### ISSUE TITLE
Config/settings file for GPflow 

### ISSUE THREAD

**Issue by alexggmatthews** [GPflow/GPflow#190]
We're beginning to find use cases for a conifg file for GPflow. Possible uses include:

1) Controlling the floating point data type as in #185 
2) Switching on a profiler as in #174 
3) Controlling the level of gitter. 

Of course all of these things could be specified at run time, but we should consider if there are benefits to having a config file.

**Comment by markvdw** [comment]
Theano has a nice interface that allows setting these things in a config file, but allowing them to be overridden in the code.

I would suggest having a module that reads the config file at import, which has local variables that can be updated at runtime, and that all places query the module.

Another suggestion of what could be added, is the verbosity of GPflow, for things like model compilation and optimisation messages.

**Comment by jameshensman** [comment]
How much do we want to be able to change the settings at run time?

I note that theano has a 500 line file just for handling settings! I think that's unmaintainable for us, I think we should use python build in configparser, and the standard python config file format. 

https://docs.python.org/3/library/configparser.html

My gut feeling is that the settings should *only* be specified in the config file, unless there's a good use case against? Simplicity where possible, right?

**Comment by jameshensman** [comment]
Another query: where should the config file live? My guess is that it should be in $HOME/.gpflowrc, and if not found, revert to the GPflow directory (via __file__). So users with installations can have a config file if they need to, but the defaults are sufficiently sensible that most users won't want to touch it.

**Comment by jameshensman** [comment]
just fyi, @alexggmatthews and @markvdw , I'm going to whip up a quick PR now so that we can discuss a concrete option.

**Comment by markvdw** [comment]
Use cases for runtime configurable settings:
- It allows examples to be guaranteed to work as expected
- Allows you to change settings temporarily during a run. It would be a bit of a hack, but it could be useful while developing.

I didn't expect it to be very difficult. I was thinking of having one module level variable for each setting, which was loaded from the config file on startup. Then functions could modify these module level variables at runtime.

But I can experiment with another PR once you've submitted an option!

**Comment by jameshensman** [comment]
made a PR: #191

**Comment by markvdw** [comment]
I just extended this to allow modification of settings on the fly. Either by direct over-writing, or using a context manager:
```
with GPflow.settings.temp_settings(config):
    # Do stuff
    pass
```

**Comment by markvdw** [comment]
One suggestion as to where this would be useful, is in the tests where float32 is creating an issue. For a small critical code path, one could choose to temporarily disable float32.

**Comment by jameshensman** [comment]
fixed in #210
